src/hist/hist2root.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+import math
+
+import ROOT
+from coffea import processor
+
+import hist.basehist
+
+
+def convert(hist: hist.basehist.BaseHist, hname: str = "") -> ROOT.TH1D | ROOT.TH2D:
+    """Convert a 1-dimensional or 2-dimensional `Hist` object or a counter to ROOT.TH1D or ROOT.TH2D
+    Parameters
+    ----------
+        hist : Hist
+            A 1-dimensional or 2-dimensional histogram object or a counter
+        hname: new name for the output histogram
+    Returns
+    -------
+        out
+            A ``ROOT.TH1D`` or ``ROOT.TH2D`` object
+    Examples
+    --------
+    Saving histograms in the coffea result into a ROOT file
+
+        result = processor.run_uproot_job(
+            samples,
+            "Events",
+            Analysis(),
+            processor.iterative_executor,
+            {"schema": NanoAODSchema},
+        )
+
+        import ROOT
+        import hist2root
+
+        fOUT = ROOT.TFile.Open('output.root','RECREATE')
+        for k in result.keys():
+            hist2root.convert(result[k],k).Write()
+        fOUT.Close()
+    """
+    if isinstance(hist, processor.defaultdict_accumulator):
+        return convertCounter(hist, hname)
+
+    if len(hist.axes) == 1:
+        return convert2TH1D(hist, hname)
+
+    if len(hist.axes) == 2:
+        return convert2TH2D(hist, hname)
+
+    raise TypeError("Not a known ROOT type")
+
+
+def convertCounter(counter: hist.basehist.BaseHist, hname: str):
+    n = len(counter.keys())
+    out = ROOT.TH1D(hname, "", n, 0, n)
+    for i, key in enumerate(counter.keys()):
+        out.SetBinContent(i + 1, counter[key])
+        out.GetXaxis().SetBinLabel(i + 1, key)
+    return out
+
+
+def convert2TH1D(hist: hist.basehist.BaseHist, hname: str) -> ROOT.TH1D:
+    name = hist.label
+    if len(name) > 0:
+        name = hname
```

Comment:
This is confusing. If this has an empty `hist.label`, then it uses that. If it's not empty, then it uses the passed name instead, even if that is empty. I really don't think that's what you intended. Instead of this structure, which is easy to get wrong, try this:

```suggestion
    name = hname or hist.label
```

That will set hname if not empty, otherwise it will try to use hist.label. Or maybe you meant the reverse, and `hname` was a fallback - in that case, `name = hist.label or hname`.